subsys/bluetooth/host/hci_core.c

```diff
@@ -2153,6 +2183,33 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 
 	bt_conn_set_state(conn, BT_CONN_CONNECTED);
 
+#if defined(CONFIG_BT_USER_PHY_UPDATE)
+	if (IS_ENABLED(CONFIG_BT_EXT_ADV) &&
+	    BT_FEAT_LE_EXT_ADV(bt_dev.le.features)) {
+		u8_t tx_phy, rx_phy;
+		int err;
+
+		err = hci_le_read_phy(conn, &tx_phy, &rx_phy);
+		if (err) {
+			BT_WARN("Failed to read PHY (%d)", err);
+		} else if (!(tx_phy == conn->le.phy.tx_phy &&
+			     rx_phy == conn->le.phy.rx_phy)) {
+			conn->le.phy.tx_phy = tx_phy;
+			conn->le.phy.rx_phy = rx_phy;
+
+			notify_le_phy_updated(conn);
```

Comment:
This is wasteful. The information is already available to the application if you move the `bt_conn_set_state()` below this code block. Plus, it's misleading: no PHY update has been performed, instead the connection was already established on a particular PHY.